### PR TITLE
adding support for setting serviceAccountName

### PR DESCRIPTION
**Pull request by byxorna**
# What and why?

Adding support to inject serviceaccounts! If a service account is already present in the pod, skip overwriting it.

NOTE: this will not work for you if you are on versions < 1.15.0 of Kubernetes, due to https://github.com/kubernetes/kubernetes/pull/78080

# Testing Steps

- [x] Added unit tests for this feature (`make test`)

# Reviewers

Required reviewers: `@byxorna`
Request reviews from other people you want to review this PR in the "Reviewers" section on the right.

> :warning: this PR must have at least 2 thumbs from the [MAINTAINERS.md](/MAINTAINERS.md) of the project before merging!

### CODE REVIEW

File: pkg/server/webhook.go [hunk 1]
```diff
@@ -420,6 +430,10 @@ func createPatch(pod *corev1.Pod, inj *config.InjectionConfig, annotations map[s
 	patch = append(patch, addContainers(pod.Spec.InitContainers, mutatedInjectedInitContainers, "/spec/initContainers")...)
 	patch = append(patch, addHostAliases(pod.Spec.HostAliases, inj.HostAliases, "/spec/hostAliases")...)
 	patch = append(patch, addVolumes(pod.Spec.Volumes, inj.Volumes, "/spec/volumes")...)
+	if inj.ServiceAccountName != "" && pod.Spec.ServiceAccountName == "" {
+		// only override the serviceaccount name if not set in the pod spec
```
Comment (roymarantz) [inline pkg/server/webhook.go]: is this documented some place beside this code spot?

Comment (byxorna) [reply pkg/server/webhook.go]: Ill update the docs as well. Good catch

File: docs/sidecar-configuration-format.md [hunk 2]
```diff
@@ -75,6 +83,6 @@ In order for the injector to know about a sidecar configuration, you need to eit
 
 1. Create a new InjectionConfiguration `yaml`
   1. Specify your `name:`. This is what you will request with `injector.tumblr.com/request=$name`
-  2. Fill in the `containers`, `volumes`, `volumeMounts`, `hostAliases`, `initContainers` and `env` fields with your configuration you want injected
+  2. Fill in the `containers`, `volumes`, `volumeMounts`, `hostAliases`, `initContainers`, `serviceAccountName`, and `env` fields with your configuration you want injected
```
Comment (roymarantz) [inline docs/sidecar-configuration-format.md]: do you want to mention that `serviceAccountName` won't override if it exists, but other will?

**Comment by byxorna** [comment]
@komapa @alex-laties re-RFR please!

**Pull request merged**